gemini/use-cases/retrieval-augmented-generation/qna_using_query_routing/utils/intent_routing.py

```diff
@@ -0,0 +1,413 @@
+        deployed_index_id: str,
+        config_file: str = "config.ini",
+        logger=logging.getLogger(),
+    ) -> None:
+        self.config = configparser.ConfigParser()
+        self.config.read(config_file)
+
+        self.logger = logger
+        self.model = model
+        self.index_endpoint = index_endpoint
+        self.deployed_index_id = deployed_index_id
+
+        self.genai_qna = QnAVectorSearch(
+            model=model,
+            index_endpoint=index_endpoint,
+            deployed_index_id=deployed_index_id,
+            config_file=config_file,
+            logger=logger,
+        )
+
+        self.genai_qna_parameters = GenerationConfig(
+            temperature=float(self.config["genai_qna"]["temperature"]),
+            max_output_tokens=int(self.config["genai_qna"]["max_output_tokens"]),
+            top_p=float(self.config["genai_qna"]["top_p"]),
+            top_k=int(self.config["genai_qna"]["top_k"]),
+        )
+
+        self.genai_chat_parameters = GenerationConfig(
+            temperature=float(self.config["genai_chat"]["temperature"]),
+            max_output_tokens=int(self.config["genai_chat"]["max_output_tokens"]),
+        )
+
+    def greetings(self, text: str) -> str:
+        """
+        Generates a friendly greeting message in response to a 'WELCOME' intent.
+
+        Leverages the provided chat model to generate a greeting tailored to the enabled programming languages.
+
+        Args:
+            text (str): The user's original message.
+
+        Returns:
+            str: The generated greeting message.
+        """
+
+        enabled_programming_language_list = self.config["default"][
+            "enabled_programming_language"
+        ].split(",")
+        enabled_programming_language = ", ".join(
+            [lang.title() for lang in enabled_programming_language_list]
+        )
+
+        chat = self.model.start_chat()
+        response = chat.send_message(
+            f"""You are Generative AI powered genai Learning Assistant.
+
+        You are trained ONLY to answers questions related to following programming languages: {enabled_programming_language}
+
+        Write a brief greeting message:"""
+        )
+        parameters_local = GenerationConfig(
+            temperature=0.7,
+            max_output_tokens=int(self.config["genai_chat"]["max_output_tokens"]),
+        )
+
+        response = chat.send_message(f"""{text}""", generation_config=parameters_local)
+        message = response.text
+        return message
+
+    def closing(self, text: str) -> str:
+        """
+        Generates a closing/thank you message in response to a 'CLOSE' intent.
+
+        Leverages the provided chat model to generate a closing message.
+
+        Args:
+            text (str): The user's original message.
+
+        Returns:
+            str: The generated closing message.
+        """
+
+        parameters_local = GenerationConfig(
```

Comment:
Similar comment as above

Reply:
The temperature was set to 0.7 for WELCOME, CLOSE intents to add creativity in response from agent, but the temperature was set to 0.3 (or lower) for WRITE_CODE intent. Hence the parameters were locally assigned the temperature value as needed by the function (intent).

Updated the parameter initialisation and now set to 0.5 (set in config.ini file) for all the intents.